src/NLog/Targets/ConsoleTarget.cs

```diff
@@ -230,27 +238,130 @@ protected override void Write(LogEventInfo logEvent)
+                            targetBufferPosition += targetBuilder.Result.Length;
+                            logEvents[i].Continuation(null);
+                        }
+                    }
+                    finally
+                    {
+                        if (targetBufferPosition > 0)
+                        {
+                            WriteBufferToOuput(output, targetBuffer.Result, targetBufferPosition);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                base.Write(logEvents);
+            }
         }
 
         /// <summary>
         /// Write to output
         /// </summary>
-        /// <param name="textLine">text to be written.</param>
-        private void WriteToOutput(string textLine)
+        private void RenderToOutput(Layout layout, LogEventInfo logEvent)
         {
             if (_pauseLogging)
             {
                 return;
             }
 
             var output = GetOutput();
+            if (WriteBuffer)
+            {
+                using (var targetBuffer = _reusableEncodingBuffer.Allocate())
+                using (var targetBuilder = ReusableLayoutBuilder.Allocate())
+                {
+                    int environmentNewLineLength = System.Environment.NewLine.Length;
+                    layout.RenderAppendBuilder(logEvent, targetBuilder.Result);
+                    if (targetBuilder.Result.Length > targetBuffer.Result.Length - environmentNewLineLength)
+                    {
+                        WriteLineToOuput(output, targetBuilder.Result.ToString());
+                    }
+                    else
+                    {
+                        targetBuilder.Result.Append(System.Environment.NewLine);
+                        targetBuilder.Result.CopyTo(0, targetBuffer.Result, 0, targetBuilder.Result.Length);
+                        WriteBufferToOuput(output, targetBuffer.Result, targetBuilder.Result.Length);
+                    }
+                }
+            }
+            else
+            {
+                WriteLineToOuput(output, RenderLogEvent(layout, logEvent));
+            }
+        }
+
+        private void WriteLineToOuput(TextWriter output, string message)
+        {
+            try
+            {
+                ConsoleTargetHelper.WriteLineThreadSafe(output, message, AutoFlush);
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                //this is a bug and therefor stopping logging. For docs, see PauseLogging property
+                _pauseLogging = true;
+                InternalLogger.Warn(ex, "An IndexOutOfRangeException has been thrown and this is probably due to a race condition." +
+                                        "Logging to the console will be paused. Enable by reloading the config or re-initialize the targets");
+            }
+            catch (ArgumentOutOfRangeException ex)
```

Comment:
proposal: catch with `when`

Reply:
No opinion on this. You are welcome to make the change.

Reply:
Made the change myself now.  Since NetCore3 now also throws OverflowException, when race-condition occurs.

Reply:
Thanks! Bit hard for me to find time. 

Will release soon.